src/send-message.ts

```diff
@@ -0,0 +1,176 @@
+import { getInput, info, setFailed } from "@actions/core";
+import { Err, Ok, Result } from "ts-results";
+import { AsyncResultWrapper } from "ts-async-results";
+import { type Client, default as zulipInit } from "zulip-js";
+
+type StringOrNumber = string | number;
+type DestinationDetails = PrivateDestination | StreamDestination;
+type PrivateDestination = {
+  kind: DestinationKind.Private;
+  destination: StringOrNumber[];
+};
+type StreamDestination = {
+  kind: DestinationKind.Stream;
+  destination: StringOrNumber;
+  topic: string;
+};
+
+enum DestinationKind {
+  Private = "private",
+  Stream = "stream",
+}
+
+function assertExhaustiveSwitch(_: never): never {
+  throw new Error("switch statement not exhaustive");
+}
+
+function allNumeric(candidate: string | string[]): boolean {
+  const oneOrMoreNumbersOnly = /^\d+$/;
+  return typeof candidate === "string"
+    ? !!oneOrMoreNumbersOnly.test(candidate)
+    : candidate.every((item: string) => oneOrMoreNumbersOnly.exec(item));
+}
+
+function parsePrivateMessageDestinations(
+  input: string
+): PrivateDestination["destination"] {
+  let rawDestinations = input.split(",");
+  return allNumeric(rawDestinations)
+    ? rawDestinations.map((item: string) => Number.parseInt(item))
+    : rawDestinations;
+}
+
+function parseStreamDestination(
+  input: string
+): StreamDestination["destination"] {
+  return allNumeric(input) ? Number.parseInt(input) : input;
+}
+
+function parseDestinationDetails({
+  kind,
+  destination,
+}: {
+  kind: DestinationKind;
+  destination: string;
+}): Result<DestinationDetails, string> {
+  switch (kind) {
+    case DestinationKind.Private: {
+      return Ok({
+        kind,
+        destination: parsePrivateMessageDestinations(destination),
+      });
+    }
+
+    case DestinationKind.Stream: {
+      return getMandatoryInputFromJob("topic")
+        .map((topic) => {
+          return {
+            kind,
+            topic,
+            destination: parseStreamDestination(destination),
+          };
+        })
+        .mapErr(() => 'topic is mandatory when type is "stream"');
+    }
+
+    default: {
+      assertExhaustiveSwitch(kind);
+    }
```

Comment:
In case someone reviewing isn't a TypeScript person: this trick will detect any unhandled (or, coincidentally, un-terminated) branches of the enum switch and turn them into compiler errors (familiar to Rust folks, probably other languages too), since `never` is a [bottom type](https://en.wikipedia.org/wiki/Bottom_type).